docs/learn/expert-guides/key-manager/get-controller-permissions.md

```diff
@@ -5,185 +5,169 @@ sidebar_position: 1
 
 # Get Controller Permissions
 
-:::caution Disclaimer
-
-This guide might contain outdated information and will be updated soon.
-
-:::
-
 In this guide, you will learn how to:
 
-- retrieve all the addresses that have some permissions set on a Universal Profile,
-- retrieve the permission of each of these **controller addresses**.
+- **retrieve all controller addresses** of a Universal Profile
+- **retrieve the permissions** of each controller address
 
 ![Get controller addresses](/img/standards/lsp6/lsp6-address-permissions-array.jpeg)
 
-We will use the [_erc725.js_](../../../tools/erc725js/getting-started.md) library to do this with just over 30 lines of code.
+:::tip Code repository
+
+You can find the scripts of this guide within our [`lukso-playground`](https://github.com/lukso-network/lukso-playground) repository.
+
+:::
 
 ## Setup
 
+First, we will install the 🛠️[`erc725.js`](../../../tools/erc725js/getting-started.md) library to easily query and decode the 🗂️[ERC725Y data storage](../../../standards/lsp-background/erc725.md#erc725y-generic-data-keyvalue-store) of the Universal Profile. You can then initialize a Universal Profile and connect to the LUKSO L16 network.
+
 ```bash
 npm install @erc725/erc725.js
 ```
 
-## Step 1 - setup web3.js and erc725.js
-
-The first step is to set up both and _web3.js_ and _erc725.js_, and connect to the LUKSO L16 network. We will also need the address of the Universal Profile that we want to get the controller addresses from.
-
 ```js
 import { ERC725 } from '@erc725/erc725.js';
 import LSP6Schema from '@erc725/erc725.js/schemas/LSP6KeyManager.json';
-import Web3 from 'web3';
-
-// setup
-const myUniversalProfileAddress = '0xC26508178c4D7d3Ad43Dcb9F9bb1fab9ceeD58B5';
-const RPC_ENDPOINT = 'https://rpc.testnet.lukso.network';
 
-// step 1 - setup erc725.js
-const web3 = new Web3(RPC_ENDPOINT);
+// Set up Universal Profile on the LUKSO Testnet
```

Comment:
This is not really correct - it doesn't setup anything on the testnet. Maybe more

"Instantiate erc725.js with a Universal Profile address" or smtgh makes more sense